lua/nvim-tree/actions/fs/rename-file.lua

```diff
@@ -26,20 +36,48 @@ function M.rename(node, to)
   local notify_from = notify.render_path(node.absolute_path)
   local notify_to = notify.render_path(to)
 
-  if utils.file_exists(to) then
-    notify.warn(err_fmt(notify_from, notify_to, "file already exists"))
-    return
+  -- create a folder for each path element if the folder does not exist
+  local idx = 0
+  local path_to_create = ""
+  local is_last_path_file = not to:match(utils.path_remove_trailing(to))
+
+  local num_nodes = get_num_nodes(utils.path_split(utils.path_remove_trailing(to)))
+  local is_error = false
+  for path in utils.path_split(to) do
+    idx = idx + 1
+
+    local p = utils.path_remove_trailing(path)
+    if #path_to_create == 0 and vim.fn.has "win32" == 1 then
```

Comment:
Please use `utils.is_wsl` or `utils.is_windows`

Reply:
I'm using wezterm terminal running zsh on macOS. I'll change the wsl, windows check on rename.
Should I change it in the create file?
I'll check into why this is failing :)

Reply:
> I'm using wezterm terminal running zsh on macOS. I'll change the wsl, windows check on rename. Should I change it in the create file because it is like what I've done right now? I'll check into why this is failing :)

My apologies, I didn't realise it came from copy. That's fine.

We need a big refactor / abstraction for that functionality, but Not Today.

I'd be very grateful for a future PR to do that :)